tests/servers.pm

```diff
@@ -1369,36 +1366,28 @@ sub runhttpsserver {
         $flags .= "--connect " . protoport("httpproxy");
     }
 
-    my $pid2;
-    my $httpspid;
-    my $port = 34813;
-    my %usedports = reverse %PORT;
-    for (1 .. $serverstartretries) {
-        $port += 1 + int(rand($portrange));
-        next if exists $usedports{$port};
-        my $options = "$flags --accept $port";
-
-        my $cmd = "$perl $srcdir/secureserver.pl $options";
-        ($httpspid, $pid2) = startnew($cmd, $pidfile, 15, 0);
-
-        if($httpspid <= 0 || !pidexists($httpspid)) {
-            # it is NOT alive
-            # don't call stopserver since that will also kill the dependent
-            # server that has already been started properly
-            $doesntrun{$pidfile} = 1;
-            $httpspid = $pid2 = 0;
-            next;
-        }
+    my $port = getfreeport($ipvnum);
+    my $options = "$flags --accept $port";
+    my $cmd = "$perl $srcdir/secureserver.pl $options";
+    my ($httpspid, $pid2) = startnew($cmd, $pidfile, 15, 0);
 
-        $doesntrun{$pidfile} = 0;
-        # we have a server!
-        if($verb) {
-            logmsg "RUN: $srvrname server is PID $httpspid port $port\n";
-        }
-        last;
+    if($httpspid <= 0 || !pidexists($httpspid)) {
+        # it is NOT alive
+        # don't call stopserver since that will also kill the dependent
+        # server that has already been started properly
+        $doesntrun{$pidfile} = 1;
+        $httpspid = $pid2 = 0;
+        logmsg "RUN: failed to start the $srvrname server\n";
+        return;
```

Comment:
This new error return path returns nothing here, while callers expect there to be 4 return values. I believe the missing first argument especially will cause an issue since it's not indicating an error. This goes for at last time call but probably the other similar bare return elsewhere in this PR.

Reply:
I figured without return values they would all be nothing or equal zero, so they would be caught by the calling function as an error. Is that wrong?